library/core/src/intrinsics.rs

```diff
@@ -1901,11 +1901,21 @@ pub unsafe fn copy_nonoverlapping<T>(src: *const T, dst: *mut T, count: usize) {
 /// ```
 /// use std::ptr;
 ///
+/// /// # Safety:
+/// /// * `ptr` must be correctly aligned for its type and non-zero.
+/// /// * `ptr` must be valid for reads of `elts` contiguous objects of type `T`.
+/// /// * Those elements must not be used after calling this function unless `T: Copy`.
```

Comment:
Is this required?

Reply:
Yes, if these are not `Copy` you'll have a double-free. `ptr::read` does not drop _or_ move the source buffer.

Reply:
Okay, I guess that's true but should this be part of safety? I think it could be but I wonder if this should be here.